Code/Tools/FBuild/FBuildCore/BFF/BFFTemplate.h

```diff
@@ -0,0 +1,36 @@
+// BFFTemplate - Lazily evalues inner scope
+//------------------------------------------------------------------------------
+#pragma once
+#ifndef FBUILD_BFFTEMPLATE_H
+#define FBUILD_BFFTEMPLATE_H
+
+// Includes
+//------------------------------------------------------------------------------
+#include "Core/Containers/Array.h"
+#include "BFFIterator.h"
+
+// Forward Declarations
+//------------------------------------------------------------------------------
+class AString;
+
+// BFFTemplate
+//------------------------------------------------------------------------------
+class BFFTemplate
+{
+public:
+	BFFTemplate();
+	BFFTemplate(const BFFTemplate& rhs);
+	explicit BFFTemplate(const BFFIterator& defStart, const BFFIterator& defEnd);
+	~BFFTemplate();
+
+	const BFFIterator & DefStart() const { return m_DefStart; }
+	const BFFIterator & DefEnd() const { return m_DefEnd; }
+
+private:
+	BFFIterator m_DefStart;
+	BFFIterator m_DefEnd;
+};
```

Comment:
This is not safe. A variable (and therefore a template) can be declared in an included file, and used after that file has been closed and has fallen out of scope. We'd need to take a copy of the BFF config segement to avoid dereferencing freed memory.